kbs/src/admin/mod.rs

```diff
@@ -2,57 +2,86 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 
-use actix_web::{http::header::Header, HttpRequest};
-use actix_web_httpauth::headers::authorization::{Authorization, Bearer};
-use config::AdminConfig;
-use jwt_simple::{
-    claims::NoCustomClaims,
-    common::VerificationOptions,
-    prelude::{Ed25519PublicKey, EdDSAPublicKeyLike},
-};
-
-pub mod config;
+use actix_web::HttpRequest;
+use log::{info, warn};
+use serde::Deserialize;
+use std::sync::Arc;
+
+pub mod allow_all;
+pub mod deny_all;
+pub mod simple;
+
 pub mod error;
 pub use error::*;
-use log::warn;
 
-#[derive(Default, Clone)]
-pub struct Admin {
-    public_key: Option<Ed25519PublicKey>,
-}
+use allow_all::InsecureAllowAllBackend;
+use deny_all::DenyAllBackend;
+use simple::{SimpleAdminBackend, SimpleAdminConfig};
 
-impl TryFrom<AdminConfig> for Admin {
-    type Error = Error;
+#[derive(Clone)]
+pub(crate) struct Admin {
+    backend: Arc<dyn AdminBackend>,
+}
 
-    fn try_from(value: AdminConfig) -> Result<Self> {
-        if value.insecure_api {
-            warn!("insecure admin APIs are enabled");
-            return Ok(Admin::default());
-        }
+// create a simple backend
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub enum AdminBackendType {
+    Simple(SimpleAdminConfig),
+    InsecureAllowAll,
+    DenyAll,
+}
 
-        let key_path = value.auth_public_key.ok_or(Error::NoPublicKeyGiven)?;
-        let user_public_key_pem = std::fs::read_to_string(key_path)?;
-        let key = Ed25519PublicKey::from_pem(&user_public_key_pem)?;
-        Ok(Self {
-            public_key: Some(key),
+impl Default for AdminBackendType {
+    fn default() -> Self {
+        AdminBackendType::Simple(SimpleAdminConfig {
+            personas: Vec::new(),
         })
     }
 }
 
-impl Admin {
-    pub(crate) fn validate_auth(&self, request: &HttpRequest) -> Result<()> {
-        let Some(public_key) = &self.public_key else {
-            return Ok(());
-        };
+#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
+#[serde(default)]
+pub struct AdminConfig {
+    pub admin_backend: AdminBackendType,
```

Comment:
With this change (2/2)
```suggestion
    #[serde(flatten)]
    pub admin_backend: AdminBackendType,
```

Furthurmore, the config can be generated into 
```toml
[admin]
type = "Simple"

[[admin.personas]]
id = "admin1"
public_key_path = "/opt/confidential-containers/trustee/admin1-pubkey.pem"
```

where wo do not have extra `admin_backend` layer.